Core.ApplicationServices/SystemUsage/ItSystemUsageService.cs

```diff
@@ -190,26 +190,54 @@ public ItSystemUsage GetById(int usageId)
             int fromSystemUsageId, 
             int relationId, 
             int toSystemUsageId, 
-            int? targetInterfaceId = null)
+            string changedDescription = "",
+            string changedReference = "",
+            int? toInterfaceId = null,
+            int? toContractId = null,
+            int? toFrequencyId = null)
         {
-            var sourceSystemUsage = _usageRepository.GetByKey(fromSystemUsageId);
-            if (sourceSystemUsage == null)
+            var fromSystemUsage = _usageRepository.GetByKey(fromSystemUsageId);
+            if (fromSystemUsage == null)
             {
                 return new OperationError("Source not found", OperationFailure.NotFound);
             }
 
-            if (!_authorizationContext.AllowModify(sourceSystemUsage))
+            if (!_authorizationContext.AllowModify(fromSystemUsage))
             {
                 return Result<SystemRelation, OperationError>.Failure(OperationFailure.Forbidden);
             }
 
-            Maybe<ItSystemUsage> targetSystemUsage = _usageRepository.GetByKey(toSystemUsageId);
-            if (targetSystemUsage.IsNone)
+            var toSystemUsage = _usageRepository.GetByKey(toSystemUsageId);
+            if (toSystemUsage == null)
             {
                 return new OperationError("Target system usage not found", OperationFailure.BadInput);
             }
 
-            var result = sourceSystemUsage.ModifyUsageRelation(_userContext.UserEntity, relationId, targetSystemUsage.Value, targetInterfaceId);
+            var toContract = Maybe<ItContract>.None;
+            if (toContractId.HasValue)
+            {
+                toContract = _contractRepository.GetById(toContractId.Value);
+                if (!toContract.HasValue)
+                {
+                    return new OperationError("Contract id does not point to a valid contract", OperationFailure.BadInput);
+                }
+
+            }
+
+            var toFrequency = Maybe<RelationFrequencyType>.None;
+            if (toFrequencyId.HasValue)
+            {
+                toFrequency = _frequencyService
+                    .GetAvailableOptions(fromSystemUsage.OrganizationId)
+                    .FirstOrDefault(x => x.Id == toFrequencyId.Value);
+
+                if (!toFrequency.HasValue)
```

Comment:
i stedet for !HasValue kan du bruge IsNone

Reply:
Copy theft fra AddRelation ;) Har nu rettet begge steder, men det er i disse to metoder jeg også mener vi skal DRY'e det lidt op .. når nu vi får talt tingene lidt på plads

Reply:
Ok vi lader den lige stå så vi ikke glemmer den i dét rul